src/test/java/uk/gov/digital/ho/pttg/api/HmrcResourceTest.java

```diff
@@ -119,4 +122,37 @@ public void shouldLogResponseSuccess() {
                     ;
         }));
     }
+
+    @Test
+    public void shouldLogRetryCountIfPassed() {
+        int someRetryCount = 2;
+        when(mockRequestHeaderData.retryCount()).thenReturn(someRetryCount);
+
+        hmrcResource.getHmrcData(new IncomeDataRequest(FIRST_NAME, LAST_NAME, NINO, DATE_OF_BIRTH, FROM_DATE, TO_DATE, ALIAS_SURNAMES));
+
+        verify(mockAppender).doAppend(argThat(argument -> {
+            LoggingEvent loggingEvent = (LoggingEvent) argument;
+
+            return loggingEvent.getFormattedMessage().equals("Retry count.") &&
+                    loggingEvent.getLevel().equals(Level.INFO) &&
+                    loggingEvent.getArgumentArray()[0].equals(new ObjectAppendingMarker("event_id", HMRC_RETRY_EVENT)) &&
+                    loggingEvent.getArgumentArray()[1].equals(new ObjectAppendingMarker(RETRY_COUNT_HEADER, someRetryCount));
+        }));
+    }
+
+    @Test
+    public void shouldNotLogRetryCountIfNotPassed() {
+        int someRetryCount = -1;
+        when(mockRequestHeaderData.retryCount()).thenReturn(someRetryCount);
+
+        hmrcResource.getHmrcData(new IncomeDataRequest(FIRST_NAME, LAST_NAME, NINO, DATE_OF_BIRTH, FROM_DATE, TO_DATE, ALIAS_SURNAMES));
+
+        verify(mockAppender, times(0)).doAppend(argThat(argument -> {
+            LoggingEvent loggingEvent = (LoggingEvent) argument;
+
+            return loggingEvent.getFormattedMessage().equals("Retry count.") &&
```

Comment:
Because this is a negative test - and they're always the least reliable - I'd favour reducing the specificity of the matcher. Just asserting that 'Retry count.' is not seen in any logs, for example, would protect us better against refactoring the code and missing out this test.